hrreport/CRM/HRReport/Form/Activity/HRAbsenceCalendar.php

```diff
@@ -216,8 +216,14 @@ function where($sourceRecordIds = null) {
           }
           if (array_key_exists("{$fieldName}_value", $this->_params)) {
             if ($field['name'] == 'activity_type_id' && count($this->_params["{$fieldName}_value"])) {
-              $sqlOp = $this->getSQLOperator(CRM_Utils_Array::value("{$fieldName}_op", $this->_params));
-              $clause = "absence.{$fieldName} {$sqlOp} (" . implode(',',$this->_params["{$fieldName}_value"]) . ") ";
+              // If the activity_type_id_value is not an array that's mean it is Not
+              // coming from the (absence type) select filter and it should be discarded.
+              if (!is_array($this->_params["{$fieldName}_value"])) {
```

Comment:
Are we sure that there is no possibility that activity_type_id_value is int / string where there is only one value?
In this case maybe it would be better to do:
(array)$this->_params["{$fieldName}_value"]
which would convert any value into array.